app/pods/publications/publication/publishpreview/controller.js

```diff
@@ -1,19 +1,122 @@
 import Controller from '@ember/controller';
-import { action } from '@ember/object';
+import {
+  action, set
+} from '@ember/object';
+import { inject as service } from '@ember/service';
 import { tracked } from '@glimmer/tracking';
+import CONFIG from 'fe-redpencil/utils/config';
+import { A } from '@ember/array';
+import moment from 'moment';
 
 export default class PublicationPublishPreviewController extends Controller {
+  // Services.
+  @service activityService;
+  @service subcasesService;
+
   // properties for making the design
   @tracked withdrawn = true;
+  @tracked testDate = new Date();
+  @tracked panelCollapsed = false;
+  @tracked showLoader = false;
+  @tracked showpublicationModal = false;
+  @tracked panelIcon = 'chevron-down'
+  @tracked publicationActivity = {
+    previewActivity: {},
+    mailContent: '',
+    subjectContent: '',
+    pieces: A([]),
+  };
 
   get publishPreviewActivities() {
     const publishPreviewActivities = this.model.publishPreviewActivities.map((activity) => activity);
     return publishPreviewActivities;
   }
 
+  /** BS PUBLICATION ACTIVITIES **/
+
   @action
-  async cancelExistingPublishPreviewActivity() {
-    alert('this action is implemented in another ticket');
+  async requestPublicationModal(activity) {
+    this.publicationActivity.pieces = activity.usedPieces;
```

Comment:
Moet je hier niet `awaiten` omdat `usedPieces` een relatie is?

Reply:
Neen in de route zit er "include: 'type,subcase,used-pieces,published-by,publishes',".
Maar ik wil er een await voor zetten als jij dat liever hebt

Reply:
Heb er eentje bij gezet